primitives/primproc/batchprimitiveprocessor.cpp

```diff
@@ -1384,8 +1392,18 @@ void BatchPrimitiveProcessor::executeTupleJoin()
             //else
             // cout << "j != joinerCount\n";
         }
+        // If we've accumulated more than 1048576 (2^20)_ of resultCounts, cut off processing. The caller will
+        // restart to continue where we left off.
+        if (resultCount >= 1048576)
+        {
+            newStartRid += newRowCount;
+            break;
+        }
     }
 
+    if (resultCount < 1048576)
```

Comment:
Why did we add this condition and what is the rationale behind the literal?

Reply:
newStartRid is returned to Execute() to tell where to pick up next time around. If resultCount < {our arbitrary cutoff count}, then return 0 to tell Execute() that we have processed all there is.

The literal must be the same as the one in the above write up. It could be a const if that matters to you. But it is arbitrary and chosen as a "reasonable" data set size.